share/picongpu/tests/XrayScattering/include/picongpu/param/species.param

```diff
@@ -0,0 +1,88 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *
+ * Forward declarations for speciesDefinition.param in case one wants to use
+ * the same particle shape, interpolation, current solver and particle pusher
+ * for all particle species.
+ */
+
+#pragma once
+
+#include "picongpu/particles/shapes.hpp"
+#include "picongpu/algorithms/FieldToParticleInterpolationNative.hpp"
+#include "picongpu/algorithms/FieldToParticleInterpolation.hpp"
+#include "picongpu/algorithms/AssignedTrilinearInterpolation.hpp"
+#include "picongpu/particles/flylite/NonLTE.def"
+#include "picongpu/fields/currentDeposition/Solver.def"
+
+
+namespace picongpu
+{
+
+/** Particle Shape definitions
+ *  - particles::shapes::CIC : 1st order
+ *  - particles::shapes::TSC : 2nd order
+ *  - particles::shapes::PCS : 3rd order
+ *  - particles::shapes::P4S : 4th order
+ *
+ *  example: using UsedParticleShape = particles::shapes::CIC;
+ */
+using UsedParticleShape = particles::shapes::Counter;
+
+/** define which interpolation method is used to interpolate fields to particles
+ */
+using UsedField2Particle = FieldToParticleInterpolation<
+    UsedParticleShape,
+    AssignedTrilinearInterpolation
+>;
+
+/** select current solver method
+ * - currentSolver::Esirkepov< SHAPE > : particle shapes - CIC, TSC, PCS, P4S (1st to 4th order)
+ * - currentSolver::VillaBune<>        : particle shapes - CIC (1st order) only
+ * - currentSolver::EmZ< SHAPE >       : particle shapes - CIC, TSC, PCS, P4S (1st to 4th order)
+ *
+ * For development purposes:
+ * - currentSolver::currentSolver::EsirkepovNative< SHAPE > : generic version of currentSolverEsirkepov
+ *   without optimization (~4x slower and needs more shared memory)
+ */
+using UsedParticleCurrentSolver = currentSolver::Esirkepov< UsedParticleShape >;
+
+/** particle pusher configuration
+ *
+ * Defining a pusher is optional for particles
+ *
+ * - particles::pusher::Vay : better suited relativistic boris pusher
+ * - particles::pusher::Boris : standard boris pusher
+ * - particles::pusher::ReducedLandauLifshitz : 4th order RungeKutta pusher
+ *                                              with classical radiation reaction
+ *
+ * For diagnostics & modeling: ------------------------------------------------
+ * - particles::pusher::Acceleration : Accelerate particles by applying a constant electric field
+ * - particles::pusher::Free : free propagation, ignore fields
+ *                             (= free stream model)
+ * - particles::pusher::Photon : propagate with c in direction of normalized mom.
+ * - particles::pusher::Probe : Probe particles that interpolate E & B
+ * For development purposes: --------------------------------------------------
+ * - particles::pusher::Axel : a pusher developed at HZDR during 2011 (testing)
+ */
+using UsedParticlePusher = particles::pusher::Boris;
+
+} // namespace picongpu
```

Comment:
Is there a difference to the default case, if not please remove the file?

Reply:
I'm setting here the particle shape to the 0th order `Counter`.

Reply:
Wait, I was not even aware we have this shape. Perhaps (separately from this PR ofc) it should be in the list of shaped in the comments above the `UsedParticleShape `

Reply:
Yes we have it, It gives you these clean edges on grid-like density.  Great for testing saxs :+1: